### PR TITLE
Cleanup *sql.DB handling to avoid intermittent C/I failures

### PR DESCRIPTION
**Pull request by jacobhaven**
r? @oconnor663 @maxtaco

### CODE REVIEW

File: daemons/gregord/flow_test.go [hunk 1]
```diff
@@ -81,34 +81,14 @@ func maybeSleep() {
...
 	srv.SetEventHandler(e)
 
-	consumer, err := newConsumer(u, rpc.SimpleLogOutput{})
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	ms := newMainServer(&opts, srv)
+	consumer := newConsumer(db, rpc.SimpleLogOutput{})
+	ms := newMainServer(&Options{BindAddress: "127.0.0.1:0"}, srv)
```
Comment (jacobhaven) [inline daemons/gregord/flow_test.go]: `mainServer` only uses `opts.BindAddress` and `opts.TLSConfig`, so we only need to set those in the test.

File: daemons/config.go [hunk 2]
```diff
@@ -139,11 +140,11 @@ func (v *FMPURIGetter) String() string {
 type DSNGetter struct {
 	S      string
 	S3conf *S3Config
-	val    *url.URL
+	val    string
```
Comment (jacobhaven) [inline daemons/config.go]: We don't actually handle DSN's as `*url.URL`s after `storage.ForceParseTime` has been called, so returning them as strings is more convenient for actual use.

**Comment by maxtaco** [comment]
@patrickxb said he'd take a look

File: daemons/config.go (hunk 2 above)
Comment (patrickxb) [inline daemons/config.go]: what's the point of `v.S`?  and shouldn't DSNGetter.String() return v.val and not v.S?

Comment (patrickxb) [inline daemons/config.go]: I guess I don't see the utility of the FMPURIGetter, DSNGetter pattern here...can you explain it to me?  Why does `Get()` return an `interface{}`?

Comment (jacobhaven) [inline daemons/config.go]: It's not the best interface, but it's what the `flag` package expects. https://golang.org/pkg/flag/#Getter

Comment (patrickxb) [inline daemons/config.go]: Oh, I didn't realize these were for flags.  Could you add a comment?

**Comment by jacobhaven** [comment]
I was able to rerun the C/I a number of times until it failed again:
`test_db_sync.go:61: Error 1215: Cannot add foreign key constraint`
It looks like it's trying to connect to the DB and reset the schema but not able to. There might be a bug in the SQL schema.

**Comment by patrickxb** [comment]
Your dsn changes LGTM pending comment on the *Getter types as flags.

**Comment by patrickxb** [comment]
That's a mysterious error...no idea why it would be intermittent...seems like it should fail every time or not at all.

**Comment by jacobhaven** [comment]
I added a one-time DB initialization script to be run by travis before the tests (the tests no longer try to init their own DB). Also renamed `daemons` to just `bin`.

File: daemons/vendor/github.com/keybase/gregor/storage/test_db_sync.go [hunk 3]
```diff
@@ -2,69 +2,71 @@ package storage
...
-	if name == "" {
-		t.Skip("TEST_MYSQL_DSN not set")
-	}
-	// We need to have parseTime=true as one of our DSN paramenters,
-	// so if it's not there, add it on. This way, in sql_scan, we'll get
-	// time.Time values back from MySQL.
-	u, err := url.Parse(name)
-	if err != nil {
-		t.Fatalf("couldn't parse TEST_MYSQL_DSN: %v", err)
+	mysqlDSN := &daemons.DSNGetter{S: os.Getenv("MYSQL_DSN")}
```
Comment (jacobhaven) [inline daemons/vendor/github.com/keybase/gregor/storage/test_db_sync.go]: Parse DSNs with the same `DSNGetter` everywhere.

**Comment by maxtaco** [comment]
Took a quick look but once ci is green ok to merge

**Pull request merged**